src/psyclone/tests/domain/lfric/lfric_builtins_test.py

```diff
@@ -250,6 +250,34 @@ def test_builtin_operator_arg():
             in str(excinfo.value))
 
 
+def test_builtin_invalid_data_type():
+    ''' Check that we raise appropriate error if we encounter a
+    built-in that takes something other than an argument of a 'real'
+    or an 'integer' data type. '''
+    old_name = lfric_builtins.BUILTIN_DEFINITIONS_FILE[:]
+    # Change the builtin-definitions file to point to one that has
+    # various invalid definitions
+    # Define the built-in name and test file
+    test_builtin_name = "inc_a_divideby_X"
+    test_builtin_file = "15.5.4_" + test_builtin_name + "_builtin.f90"
+    lfric_builtins.BUILTIN_DEFINITIONS_FILE = \
+        os.path.join(BASE_PATH, "invalid_builtins_mod.f90")
+    _, invoke_info = parse(
```

Comment:
Please merge some lines to shorten this bit.

Reply:
Done.